### PR TITLE
Add nautilus telemetry

### PR DESCRIPTION
**Pull request by ladd**
Adding a registry entry for [NautilusTelemetry](https://github.com/eBay/NautilusTelemetry)

### CODE REVIEW

File: content/en/registry/sdk-swift-nautilustelemetry.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+tags:
+  - swift
+  - instrumentation
+  - ios
+  - macOS
+  - tvOS
+  - watchOS
+repo: https://github.com/eBay/NautilusTelemetry
+license: MIT
+description: NautilusTelemetry is an iOS-oriented Swift package to collect OpenTelemetry data and submit it in OTLP-JSON format to an OpenTelemetry Collector instance. gRPC is not currently supported in order to keep the package size as small as possible. Not all features of OpenTelemetry are supported, and this package should be considered experimental.
```
Comment (svrnm) [inline content/en/registry/sdk-swift-nautilustelemetry.md:15]: "is not currently" -> "is currently not"?

**Review by svrnm — COMMENTED** [review]
lgtm, only a small word change suggested. Leaving it to @open-telemetry/swift-approvers to verify that this is a good fit for the registry.

File: content/en/registry/sdk-swift-nautilustelemetry.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+title: NautilusTelemetry
+registryType: SDK
```
Comment (chalin) [inline content/en/registry/sdk-swift-nautilustelemetry.md:3]: AFAIK, we don't have an `sdk` registry type yet:

```console
$ find content/en/registry/*.* -exec grep -e 'registryType:' {} \; | sort | uniq
registryType: core
registryType: exporter
registryType: extension
registryType: instrumentation
registryType: processor
registryType: receiver
registryType: utilities
```

Is this a type that we want to add @austinlparker?

If so, it'll need to be in lowercase @ladd.

File: content/en/registry/sdk-swift-nautilustelemetry.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  - swift
+  - instrumentation
+  - ios
+  - macOS
+  - tvOS
+  - watchOS
+repo: https://github.com/eBay/NautilusTelemetry
+license: MIT
+description: NautilusTelemetry is an iOS-oriented Swift package to collect OpenTelemetry data and submit it in OTLP-JSON format to an OpenTelemetry Collector instance. gRPC is not currently supported in order to keep the package size as small as possible. Not all features of OpenTelemetry are supported, and this package should be considered experimental.
+
```
Comment (chalin) [inline content/en/registry/sdk-swift-nautilustelemetry.md:16]: Nit: drop the empty line

**Review by chalin — COMMENTED** [review]
See inline comments.

**Comment by austinlparker** [comment]
I have some questions about this package; Is this an implementation of the OpenTelemetry API or SDK specification, or is this something that maps existing telemetry signals into OTLP-JSON?

**Comment by ladd** [comment]
> I have some questions about this package; Is this an implementation of the OpenTelemetry API or SDK specification, or is this something that maps existing telemetry signals into OTLP-JSON?

This is an implementation of a subset of the API and SDK specification, as well as an exporter for OTLP-JSON.

I wasn't sure how best to classify it -- happy to take your advice.

**Comment by austinlparker** [comment]
Huh. Well, nominally I mark implementations as 'core', but this isn't actually owned by otel org so I'm not sure that's appropriate. 'Instrumentation', maybe? I'm pretty out of my depth when it comes to swift, what differs between this and otel-swift?

**Comment by nachoBonafonte** [comment]
From my perspective this is a OTLP (Opentelemetry Protocol) compatible Observability library, not an Opentelemetry project, and definitely not an Opentelemetry SDK. The spec clearly details what an Opentelemetry API and Opentelemetry SDK must include, and how it must be structured, and this is not written following those principles.

Opentelemetry is all about the spec, the API and the ecosystem. The idea is to have a common spec and API that make the interconnection of all those pieces easy for the user and interchangeable, from the protocol but also from the library and source level, so there is a rich ecosystem of components that can be used for everyone. This project does not work in that direction, it exports Observability data in OTLP JSON format but has none of the other properties.

Said that, I don't know what was the reason for this project to be started when there has been a working Opentelemetry distribution for Swift for quite a long time now and that is being currently used by several well-known companies in commercial products; but I am sure that there was a good reason. You must know that in the opentelemetry-swift project we are more than open to add/change/include/adapt the code to the current needs of any individual/company that is interested in using Opentelemetry with Swift: feedback, ideas, PR's and help are more than welcome.

**Review by nachoBonafonte — COMMENTED** [review]
From my perspective this is a OTLP (Opentelemetry Protocol) compatible Observability library, not an Opentelemetry project, and definitely not an Opentelemetry SDK. The spec clearly details what an Opentelemetry API and Opentelemetry SDK must include, and how it must be structured, and this is not written following those principles.

Opentelemetry is all about the spec, the API and the ecosystem. The idea is to have a common spec and API that make the interconnection of all those pieces easy for the user and interchangeable, from the protocol but also from the library and source level, so there is a rich ecosystem of components that can be used for everyone. This project does not work in that direction, it exports Observability data in OTLP JSON format but has none of the other properties.

Said that, I don't know what was the reason for this project to be started when there has been a working Opentelemetry distribution for Swift for quite a long time now and that is being currently used by several well-known companies in commercial products; but I am sure that there was a good reason. You must know that in the opentelemetry-swift project we are more than open to add/change/include/adapt the code to the current needs of any individual/company that is interested in using Opentelemetry with Swift: feedback, ideas, PR's and help are more than welcome.

**Comment by ladd** [comment]
> From my perspective this is a OTLP (Opentelemetry Protocol) compatible Observability library, not an Opentelemetry project, and definitely not an Opentelemetry SDK. The spec clearly details what an Opentelemetry API and Opentelemetry SDK must include, and how it must be structured, and this is not written following those principles.

Correct. I'm not intending to present this as an OpenTelemetry SDK, or an official OpenTelemetry project. I did not see a registry type value that appeared to match what I'm doing, but I'm happy to use whatever works for the project.

>Said that, I don't know what was the reason for this project to be started when there has been a working Opentelemetry distribution for Swift for quite a long time now and that is being currently used by several well-known companies in commercial products; but I am sure that there was a good reason

My motivation is twofold:

1. The official library is about 500K lines of source when you include all the dependencies. My project is about 3K lines, and is better suited for inclusion in an iOS native app.
2. My project is specifically oriented towards Apple platforms, while the official project is built as cross-platform swift. This changes the OS features you would naturally use: MetricKit, monotonic timers, unfair lock, etc.

**Comment by nachoBonafonte** [comment]
> The official library is about 500K lines of source when you include all the dependencies. My project is about 3K lines, and is better suited for inclusion in an iOS native app.

The project is statically linked when included in an iOS app, that means that only those bytes used from the project will be included in the final binary

> My project is specifically oriented towards Apple platforms, while the official project is built as cross-platform swift. This changes the OS features you would naturally use: MetricKit, monotonic timers, unfair lock, etc.

Being a cross-platform project does not mean it cannot use Apple technologies, in fact some of the code is only available for Apple platforms because of this, for example opentelemetry-swift project includes a MetricKit importer to convert MetricKit metrics into Opentelemetry metrics, and it only works on Apple platforms.

> I'm not intending to present this as an OpenTelemetry SDK, or an official OpenTelemetry project.

What I tried to say is that exporting OTLP doesn't convert a project into an Opentelemetry project (official or not). From my perspective, and as I tried to explain before, this is not an Opentelemetry project and should not be included in the registry.

**Comment by ladd** [comment]
> The project is statically linked when included in an iOS app, that means that only those bytes used from the project will be included in the final binary

Compilation speed is also a factor for our use case.

> What I tried to say is that exporting OTLP doesn't convert a project into an Opentelemetry project (official or not). From my perspective, and as I tried to explain before, this is not an Opentelemetry project and should not be included in the registry.

Perhaps I've missed something. The registry says:

>The OpenTelemetry Registry allows you to search for instrumentation libraries, tracer implementations, utilities, and other useful projects in the OpenTelemetry ecosystem.

I think my project is "useful", is connected to the OpenTelemetry ecosystem. If there's a more stringent criteria for inclusion, please send me a pointer -- it would be helpful to call this out at the top of the registry.

**Comment by austinlparker** [comment]
It's true that there's no requirement that a registry item be an official OpenTelemetry project. I'm afraid I lack the domain knowledge to really evaluate this debate, but I feel like the description/type is clear about what the package does. I'm going to approve and merge this based on the updated description.

**Review by austinlparker — APPROVED** [review]

**Pull request merged**